cmd/embedded-cluster/reset.go

```diff
@@ -0,0 +1,299 @@
+	}
+	// set up kube client
+	err = currentHost.configureKubernetesClient()
+	if err != nil {
+		return currentHost, err
+	}
+	// fetch node object
+	err = currentHost.getNodeObject(ctx)
+	if err != nil {
+		return currentHost, err
+	}
+	// control plane only stff
+	if currentHost.isControlPlane() {
+		// fetch controlNode
+		err := currentHost.getControlNodeObject(ctx)
+		if err != nil {
+			return currentHost, err
+		}
+	}
+	return currentHost, nil
+}
+
+func checkErrPrompt(err error) bool {
+	if err == nil {
+		return true
+	}
+	fmt.Println("-----")
+	fmt.Println(err)
+	fmt.Println("-----")
+	fmt.Println("An error has occured while trying to reset this node.")
+	fmt.Println("Continuing may leave the cluster in an unexpected state")
+	return prompts.New().Confirm("Do you want to continue anyway?", false)
+}
+
+var resetCommand = &cli.Command{
+	Name: "reset",
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name:   "yes-really-reset",
+			Hidden: true,
+			Value:  false,
+		},
+	},
+	Usage: "Reset the node this command is run from",
+	Action: func(c *cli.Context) error {
+
+		fmt.Println("This command will completely reset this node, removing it from the cluster")
+		if !prompts.New().Confirm("Do you want to continue?", false) {
+			fmt.Println("aborting.")
+			return nil
+		}
+
+		fmt.Println("gathering facts...")
+		// populate options struct with host information
+		currentHost, err := newHostInfo(c.Context)
+		if err != nil {
+			fmt.Println(err)
+			return nil
+		}
+
+		// basic check to see if it's safe to remove this node from the cluster
+		if currentHost.isControlPlane() {
+			safeToRemove, reason, err := currentHost.checkQuorumSafety(c)
+			if err != nil {
+				fmt.Println(err)
+				return nil
+			}
+			if !safeToRemove {
+				fmt.Println(reason)
+				fmt.Println("run reset command again with --yes-really-reset to ignore this")
+				return nil
+			}
+		}
+
+		// determine if this is the only node in the cluster
+		// if this is a single node we can skip a lot of steps
+		nodeList := corev1.NodeList{}
+		currentHost.Kclient.List(c.Context, &nodeList)
+		if len(nodeList.Items) == 1 {
+			nodeName := nodeList.Items[0].Name
+			if nodeName != currentHost.Hostname {
+				fmt.Println("detected a single node cluster, but the node's name doesn't match our hostname")
+				return nil
+			}
+			// stop k0s
+			fmt.Printf("resetting %s...\n", binName)
+			err = currentHost.stopAndResetK0s()
+			if !checkErrPrompt(err) {
+				return nil
+			}
+			return nil
+		}
+
+		// drain node
+		fmt.Println("draining node...")
+		err = currentHost.drainNode()
+		if !checkErrPrompt(err) {
+			return nil
+		}
+
+		// remove node from cluster
+		fmt.Println("removing node from cluster...")
+		err = currentHost.Kclient.Delete(c.Context, &currentHost.Node)
+		if !checkErrPrompt(err) {
+			return nil
+		}
+
+		// controller pre-reset
+		if currentHost.isControlPlane() {
+
+			// delete controlNode object from cluster
+			fmt.Println("deleting controlNode...")
```

Comment:
Capital D.

I assume this is the same controlnode as [here](https://docs.k0sproject.io/v1.28.4+k0s.0/autopilot/?h=controlnode#q-how-do-i-apply-the-plan-and-controlnode-crds). If so, let's capitalize ControlNode like k0s does.